app/components/Map.tsx

```diff
@@ -64,6 +64,18 @@ function makeStationMarker(
   const markerElement = document.createElement('div');
   markerElement.className = 'marker';
   markerElement.setAttribute('style', `width: ${diameter}px; height: ${diameter}px;`);
+  if (markerElement) {
+    markerElement.addEventListener('click', function() {
+      const activeMarkers: NodeListOf<Element> = document.querySelectorAll('.marker');
+      activeMarkers.forEach(marker => {
```

Comment:
We only need to update two Markers, but we iterate over all of them. How about this:
1. Call `.remove('active')` for all elements found with `document.querySelectorAll('.marker.active')`.
2. Call `event.currentTarget.classList.remove('active')`.

Reply:
fixed